sp-sim/src/update.rs

```diff
@@ -192,8 +233,7 @@ impl SimSpUpdate {
                         None => return Err(SpError::InvalidSlotForComponent),
                     }
                 } else {
```

Comment:
With the recent changes to the host flash mux stuff, I think we could remove this conditional entirely and just always use `slot`. (If you try this and it breaks other tests I'd definitely punt it to later work when we start wanting to test host flash updates more faithfully.)

Reply:
I'd like to better understand this one so I'll do this in a follow-on PR.